http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=675407


Deal with X 1.12 by calling vgaHWSetStdFuncs().

Thanks to Leon Winter for testing the initial patch and proposing a
working one.

Signed-off-by: Cyril Brulebois <kibi@debian.org>
--- a/src/via_driver.c
+++ b/src/via_driver.c
@@ -997,6 +997,11 @@ VIAPreInit(ScrnInfoPtr pScrn, int flags)
 #endif
     }
 
+    hwp = VGAHWPTR(pScrn);
+
+    /* Needed with X 1.12 */
+    vgaHWSetStdFuncs(hwp);
+
     if (pVia->Chipset == VIA_CLE266)
         ViaDoubleCheckCLE266Revision(pScrn);
 
@@ -1012,8 +1017,6 @@ VIAPreInit(ScrnInfoPtr pScrn, int flags)
     struct pci_device *vgaDevice = viaPciDeviceVga();
 #endif
 
-    hwp = VGAHWPTR(pScrn);
-
     switch (pVia->Chipset) {
         case VIA_CLE266:
 #ifdef XSERVER_LIBPCIACCESS
